src/redux/home/home.js

```diff
@@ -0,0 +1,91 @@
+// ------------- DUCK PATH ---------------
+const GET_DATA = 'REDUX/HOME/GET_DATA';
+const STORE_FETCHED_DATA = 'REDUX/HOME/STORE_FETCHED_DATA';
+const FILTER_DATA = 'REDUX/HOME/FILTER_DATA';
+const CHANGE_LOAD_STATE = 'REDUX/HOME/CHANGE_LOAD_STATE';
+// -------------- DEFAUL STATE -----------
+const defaultState = [];
+// -------------- ACTIONS ----------------
+const getData = (payload) => ({
+  type: GET_DATA,
+  payload,
+});
+const storeFetchedData = (payload) => ({
+  type: STORE_FETCHED_DATA,
+  payload,
+});
+const filterData = (payload) => ({
+  type: FILTER_DATA,
+  payload,
+});
+const changeLoadingState = (payload) => ({
+  type: CHANGE_LOAD_STATE,
+  payload,
+});
+  // -------------REDUCERS ----------------
+const detailDataReducer = (state = defaultState, action) => {
+  switch (action.type) {
+    case GET_DATA:
+      return state;
+    case STORE_FETCHED_DATA:
+      return action.payload;
+    default:
```

Comment:
[OPTIONAL]

- I would suggest keeping the reducers, actions, and middleware in a separate file so that it will easy to handle, debug, and finding the bug in a larger project